test/number.spec.ts

```
@@ -178,14 +178,15 @@ describe('number', () => {
         }
       });
 
-      it('provides numbers with a given precision of 0.5 steps', () => {
+      it.todo('provides numbers with a given precision of 0.5 steps', () => {
```

I'm a bit lost on how to achieve this with 'fractionDigits' being present. Any math person can help me out?

---

Should this feature maybe entirely be implemented in its own pull request? I would have some pretty hacky solution to solve this, but I have a feeling it will result in a very big discussion. Thats why I'm askin wether `fractionDigits` and `steps` should be 2 separate concernces.

---

I don't think "steps" is particularly useful. I would not include it in the new method. 

Assuming we only allow integer values for fraction digits then that's equivalent to precisions of the form 10^x only.

---

> Assuming we only allow integer values for fraction digits [...]

I don't think that's the current behavior. There is also no test case for that. What would happen in your opinion if someone passed a float into `frictionDigits`?

---

You can't have a number with "two and a half
" decimal places. It should throw an error.

---

FWIW toFixed seems to just floor its parameter ie

2.1234567.toFixed(2.99) just acts like 2.1234567.toFixed(2)

---

> I don't think "steps" is particularly useful. I would not include it in the new method.

The `precision` aka `step` feature is intended to be used for or in conjunction with HTML's input step feature:

https://www.w3schools.com/tags/att_input_step.asp

> I'm a bit lost on how to achieve this with 'fractionDigits' being present.

I think it is not necessary.

> Any math person can help me out?

See also: least common multiple
I assume there also approximation possible using the log function.

> FWIW toFixed seems to just floor its parameter ie

IMO there is no need to go beyond what JS does itself.